project_euler/problem_203/sol1.py

```diff
@@ -0,0 +1,185 @@
+"""
+Project Euler Problem 203: https://projecteuler.net/problem=203
+
+The binomial coefficients (n k) can be arranged in triangular form, Pascal's
+triangle, like this:
+                            1
+                        1       1
+                    1		2       1
+                1		3		3       1
+            1		4		6		4		1
+        1		5		10		10		5		1
+    1		6		15		20		15		6		1
+1		7		21		35		35		21		7		1
+                        .........
+
+It can be seen that the first eight rows of Pascal's triangle contain twelve
+distinct numbers: 1, 2, 3, 4, 5, 6, 7, 10, 15, 20, 21 and 35.
+
+A positive integer n is called squarefree if no square of a prime divides n.
+Of the twelve distinct numbers in the first eight rows of Pascal's triangle,
+all except 4 and 20 are squarefree. The sum of the distinct squarefree numbers
+in the first eight rows is 105.
+
+Find the sum of the distinct squarefree numbers in the first 51 rows of
+Pascal's triangle.
+
+References:
+- https://en.wikipedia.org/wiki/Pascal%27s_triangle
+"""
+
+import math
+from typing import List, Set
+
+
+def get_pascal_triangle_unique_coefficients(depth: int) -> Set[int]:
+    """
+    Returns the unique coefficients of a Pascal's triangle of depth "depth".
+
+    The coefficients of this triangle are symmetric. A further improvement to this
+    method could be to calculate the coefficients once per level. Nonetheless,
+    the current implementation is fast enough for the original problem.
+
+    >>> get_pascal_triangle_unique_coefficients(1)
+    {1}
+    >>> get_pascal_triangle_unique_coefficients(2)
+    {1}
+    >>> get_pascal_triangle_unique_coefficients(3)
+    {1, 2}
+    >>> get_pascal_triangle_unique_coefficients(8)
+    {1, 2, 3, 4, 5, 6, 7, 35, 10, 15, 20, 21}
+    """
+    coefficients = set()
+    previous_coefficients = []
+    for step in range(1, depth + 1):
+        if step == 1:
+            coefficients.add(1)
+            previous_coefficients = [1]
+        else:
+            coefficients_begins_one = previous_coefficients + [0]
+            coefficients_ends_one = [0] + previous_coefficients
+            previous_coefficients = []
```

Comment:
```suggestion
    coefficients = {1}
    previous_coefficients = [1]
    for step in range(2, depth + 1):
        coefficients_begins_one = previous_coefficients + [0]
        coefficients_ends_one = [0] + previous_coefficients
        previous_coefficients = []
```
Please test whether this works or not.

Reply:
Tested locally and works. Also black, flake8 and doctests pass. Should I commit locally or use the "commit suggestion" button?